From 6131a4cfba214b89d0197510a4f4cab33a160bfd Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Mon, 28 Feb 2011 19:43:32 -0500
Subject: [PATCH] gsm_manager_get does not return a reference

This causes gnome-session to crash and burn whenever the new
if-session condition is used.
---
 gnome-session/gsm-autostart-app.c |    8 --------
 1 files changed, 0 insertions(+), 8 deletions(-)

diff --git a/gnome-session/gsm-autostart-app.c b/gnome-session/gsm-autostart-app.c
index 81e66ce..05285d7 100644
--- a/gnome-session/gsm-autostart-app.c
+++ b/gnome-session/gsm-autostart-app.c
@@ -548,8 +548,6 @@ setup_condition_monitor (GsmAutostartApp *app)
                 g_signal_connect (manager, "notify::session-name",
                                   G_CALLBACK (if_session_condition_cb), app);
                 g_free (session_name);
-
-                g_object_unref (manager);
         } else if (kind == GSM_CONDITION_UNLESS_SESSION) {
                 GsmManager *manager;
                 char *session_name;
@@ -563,8 +561,6 @@ setup_condition_monitor (GsmAutostartApp *app)
                 g_signal_connect (manager, "notify::session-name",
                                   G_CALLBACK (unless_session_condition_cb), app);
                 g_free (session_name);
-
-                g_object_unref (manager);
         } else {
                 disabled = TRUE;
         }
@@ -869,8 +865,6 @@ is_conditionally_disabled (GsmApp *app)
                 g_object_get (manager, "session-name", &session_name, NULL);
                 disabled = strcmp (session_name, key) != 0;
                 g_free (session_name);
-
-                g_object_unref (manager);
         } else if (kind == GSM_CONDITION_UNLESS_SESSION) {
                 GsmManager *manager;
                 char *session_name;
@@ -881,8 +875,6 @@ is_conditionally_disabled (GsmApp *app)
                 g_object_get (manager, "session-name", &session_name, NULL);
                 disabled = strcmp (session_name, key) == 0;
                 g_free (session_name);
-
-                g_object_unref (manager);
         } else {
                 disabled = TRUE;
         }
-- 
1.7.4.1

